src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfig.java

```diff
@@ -45,205 +45,315 @@
-		this.externalConfigLogged = externalConfigLogged;
-	}
+    /**
+     * Checks if internal config must be logged
+     * @return true/false
+     */
+    public boolean shouldLogInternalConfig() {
+        return logInternalConfig;
+    }
 
-	public boolean isEnabled() {
+    /**
+     * Checks if compliance is enabled
+     * @return true/false
+     */
+    public boolean isEnabled() {
         return this.enabled;
     }
 
-    //cached
-    @SuppressWarnings("unchecked")
-    private Set<String> getFieldsForIndex0(String index) {
+    /**
+     * Checks if logs diffs must be recorded for write requests
+     * Log metadata only for write requests must be disabled
+     * @return true/false
+     */
+    public boolean shouldLogDiffsForWrite() {
+        return !shouldLogWriteMetadataOnly() && logDiffsForWrite;
+    }
+
+    /**
+     * Checks if only metadata for write requests should be logged
+     * @return true/false
+     */
+    public boolean shouldLogWriteMetadataOnly() {
+        return logWriteMetadataOnly;
+    }
 
-        if(index == null) {
-            return Collections.EMPTY_SET;
+    /**
+     * Checks if only metadata for read requests should be logged
+     * @return true/false
+     */
+    public boolean shouldLogReadMetadataOnly() {
+        return logReadMetadataOnly;
+    }
+
+    /**
+     * Get set of immutable index pattern
+     * @return set of index patterns
+     */
+    public Set<String> getImmutableIndicesPatterns() {
+        return immutableIndicesPatterns;
+    }
+
+    /**
+     * Get the salt in bytes for filed anonymization
+     * @return salt in bytes
+     */
+    public byte[] getSalt16() {
+        return salt16.clone();
+    }
+
+    /**
+     * This function is used for caching the fields
+     * @param index index to check for fields
+     * @return set of fields which is used by cache
+     */
+    private Set<String> getFieldsForIndex(String index) {
+        if (index == null) {
+            return Collections.emptySet();
         }
 
-        if(auditLogIndex != null && auditLogIndex.equalsIgnoreCase(index)) {
-            return Collections.EMPTY_SET;
+        if (auditLogIndex != null && auditLogIndex.equalsIgnoreCase(index)) {
+            return Collections.emptySet();
         }
 
-        if(auditLogPattern != null) {
-            if(index.equalsIgnoreCase(getExpandedIndexName(auditLogPattern, null))) {
-                return Collections.EMPTY_SET;
+        if (auditLogPattern != null) {
+            if (index.equalsIgnoreCase(getExpandedIndexName(auditLogPattern, null))) {
+                return Collections.emptySet();
             }
         }
 
         final Set<String> tmp = new HashSet<String>(100);
-        for(String indexPattern: readEnabledFields.keySet()) {
-            if(indexPattern != null && !indexPattern.isEmpty() && WildcardMatcher.match(indexPattern, index)) {
+        for (String indexPattern : readEnabledFields.keySet()) {
```

Comment:
Consider:
```
        return readEnabledFields.entrySet().stream()
                .filter(entry -> WildcardMatcher.match(entry.getKey(), index))
                .flatMap(entry -> entry.getValue().stream())
                .collect(Collectors.toSet());
```
There is no point of adding `null` or `empty` `indexPattern` to `readEnabledFields`, add it as a filter to the constructor.